<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Human Protein Atlas in R • hpar</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Human Protein Atlas in R">
<meta property="og:description" content="hpar">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hpar</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.41.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/hpar.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="hpar_files/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="hpar_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="hpar_files/datatables-binding-0.26/datatables.js"></script><link href="hpar_files/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="hpar_files/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="hpar_files/dt-core-1.12.1/js/jquery.dataTables.min.js"></script><link href="hpar_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="hpar_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Human Protein Atlas in R</h1>
                        <h4 data-toc-skip class="author">Laurent
Gatto</h4>
            <address class="author_afil">
      de Duve Institute, UCLouvain, Belgium<br><div class="hidden name"><code>hpar.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The Human Protein Atlas (HPA) is a systematic study oh the
      human proteome using antibody-based proteomics. Multiple tissues
      and cell lines are systematically assayed affinity-purified
      antibodies and confocal microscopy. The <em>hpar</em> package is
      an R interface to the HPA project. It distributes three data sets,
      provides functionality to query these and to access detailed
      information pages, including confocal microscopy images available
      on the HPA web page.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<div class="section level3">
<h3 id="the-hpa-project">The HPA project<a class="anchor" aria-label="anchor" href="#the-hpa-project"></a>
</h3>
<p>From the <a href="http://www.proteinatlas.org/" class="external-link">Human Protein
Atlas</a> <span class="citation">(Uhlén et al. 2005; Uhlen et al.
2010)</span> site:</p>
<blockquote>
<p>The Swedish Human Protein Atlas project, funded by the Knut and Alice
Wallenberg Foundation, has been set up to allow for a systematic
exploration of the human proteome using Antibody-Based Proteomics. This
is accomplished by combining high-throughput generation of
affinity-purified antibodies with protein profiling in a multitude of
tissues and cells assembled in tissue microarrays. Confocal microscopy
analysis using human cell lines is performed for more detailed protein
localisation. The program hosts the Human Protein Atlas portal with
expression profiles of human proteins in tissues and cells.</p>
</blockquote>
<p>The <em><a href="https://bioconductor.org/packages/3.17/hpar" class="external-link">hpar</a></em> package
provides access to HPA data from the R interface. It also distributes
the following data sets:</p>
<p>Several flat files are distributed by the HPA project and available
within the package as data.frames, other datasets are available through
a search query on the HPA website. The description below is taken from
the HPA site:</p>
<ul>
<li><p>hpaNormalTissue: Normal tissue data. Expression profiles for
proteins in human tissues based on immunohistochemisty using tissue
micro arrays. The tab-separated file includes Ensembl gene identifier
(“Gene”), tissue name (“Tissue”), annotated cell type (“Cell type”),
expression value (“Level”), and the gene reliability of the expression
value (“Reliability”).</p></li>
<li><p>hpaNormalTissue16.1: Same as above, for version 16.1.</p></li>
<li><p>hpaCancer: Pathology data. Staining profiles for proteins in
human tumor tissue based on immunohistochemisty using tissue micro
arrays and log-rank P value for Kaplan-Meier analysis of correlation
between mRNA expression level and patient survival. The tab-separated
file includes Ensembl gene identifier (“Gene”), gene name (“Gene name”),
tumor name (“Cancer”), the number of patients annotated for different
staining levels (“High”, “Medium”, “Low” &amp; “Not detected”) and
log-rank p values for patient survival and mRNA correlation (“prognostic
- favourable”, “unprognostic - favourable”, “prognostic - unfavourable”,
“unprognostic - unfavourable”).</p></li>
<li><p>hpaCancer16.1: Same as above, for version 16.1</p></li>
<li><p>rnaConsensusTissue: RNA consensus tissue gene data. Consensus
transcript expression levels summarized per gene in 54 tissues based on
transcriptomics data from HPA and GTEx. The consensus normalized
expression (“nTPM”) value is calculated as the maximum nTPM value for
each gene in the two data sources. For tissues with multiple sub-tissues
(brain regions, lymphoid tissues and intestine) the maximum of all
sub-tissues is used for the tissue type. The tab-separated file includes
Ensembl gene identifier (“Gene”), analysed sample (“Tissue”) and
normalized expression (“nTPM”).</p></li>
<li><p>rnaHpaTissue: RNA HPA tissue gene data. Transcript expression
levels summarized per gene in 256 tissues based on RNA-seq. The
tab-separated file includes Ensembl gene identifier (“Gene”), analysed
sample (“Tissue”), transcripts per million (“TPM”), protein-transcripts
per million (“pTPM”) and normalized expression (“nTPM”).</p></li>
<li><p>rnaGtexTissue: RNA GTEx tissue gene data. Transcript expression
levels summarized per gene in 37 tissues based on RNA-seq. The
tab-separated file includes Ensembl gene identifier (“Gene”), analysed
sample (“Tissue”), transcripts per million (“TPM”), protein-transcripts
per million (“pTPM”) and normalized expression (“nTPM”). The data was
obtained from GTEx.</p></li>
<li><p>rnaFantomTissue: RNA FANTOM tissue gene data. Transcript
expression levels summarized per gene in 60 tissues based on CAGE data.
The tab-separated file includes Ensembl gene identifier (“Gene”),
analysed sample (“Tissue”), tags per million (“Tags per million”),
scaled-tags per million (“Scaled tags per million”) and normalized
expression (“nTPM”). The data was obtained from FANTOM5.</p></li>
<li><p>rnaGeneTissue21.0: RNA HPA tissue gene data. Transcript
expression levels summarized per gene in 37 tissues based on RNA-seq,
for hpa version 21.0. The tab-separated file includes Ensembl gene
identifier (“Gene”), analysed sample (“Tissue”), transcripts per million
(“TPM”), protein-transcripts per million (“pTPM”).</p></li>
<li><p>rnaGeneCellLine: RNA HPA cell line gene data. Transcript
expression levels summarized per gene in 69 cell lines. The
tab-separated file includes Ensembl gene identifier (“Gene”), analysed
sample (“Cell line”), transcripts per million (“TPM”), protein-coding
transcripts per million (“pTPM”) and normalized expression
(“nTPM”).</p></li>
<li><p>rnaGeneCellLine16.1: Same as above, for version 16.1.</p></li>
<li><p>hpaSubcellularLoc: Subcellular location data. Subcellular
location of proteins based on immunofluorescently stained cells. The
tab-separated file includes the following columns: Ensembl gene
identifier (“Gene”), name of gene (“Gene name”), gene reliability score
(“Reliability”), enhanced locations (“Enhanced”), supported locations
(“Supported”), Approved locations (“Approved”), uncertain locations
(“Uncertain”), locations with single-cell variation in intensity
(“Single-cell variation intensity”), locations with spatial single-cell
variation (“Single-cell variation spatial”), locations with observed
cell cycle dependency (type can be one or more of biological definition,
custom data or correlation) (“Cell cycle dependency”), Gene Ontology
Cellular Component term identifier (“GO id”).</p></li>
<li><p>hpaSubcellularLoc16.1: Same as above, for version 16.1.</p></li>
<li><p>hpaSubcellularLoc14: Same as above, for version 14.</p></li>
<li><p>hpaSecretome: Secretome data. The human secretome is here defined
as all Ensembl genes with at least one predicted secreted transcript
according to HPA predictions. The complete information about the HPA
Secretome data is given on <a href="https://www.proteinatlas.org/humanproteome/blood/secretome" class="external-link uri">https://www.proteinatlas.org/humanproteome/blood/secretome</a>.
This dataset has 315 columns and includes the Ensembl gene identifier
(“Gene”). Information about the additionnal variables can be found <a href="https://www.proteinatlas.org/search" class="external-link">here</a> by clicking on
<em>Show/hide columns</em>.</p></li>
</ul>
<p>The <code><a href="../reference/getHpaDate.html">hpar::allHparData()</a></code> returns a list of all datasets
(see below).</p>
</div>
<div class="section level3">
<h3 id="hpa-data-usage-policy">HPA data usage policy<a class="anchor" aria-label="anchor" href="#hpa-data-usage-policy"></a>
</h3>
<p>The use of data and images from the HPA in publications and
presentations is permitted provided that the following conditions are
met:</p>
<ul>
<li>The publication and/or presentation are solely for informational and
non-commercial purposes.</li>
<li>The source of the data and/or image is referred to the HPA site<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> and/or one
or more of our publications are cited.</li>
</ul>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.17/hpar" class="external-link">hpar</a></em> is
available through the Bioconductor project. Details about the package
and the installation procedure can be found on its <a href="http://bioconductor.org/packages/devel/bioc/html/hpar.html" class="external-link">landing
page</a>. To install using the dedicated Bioconductor infrastructure,
run :</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## install BiocManager only one</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="co">## install hpar</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"hpar"</span><span class="op">)</span></span></code></pre></div>
<p>After installation, <em><a href="https://bioconductor.org/packages/3.17/hpar" class="external-link">hpar</a></em> will
have to be explicitly loaded with</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"hpar"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## snapshotDate(): 2022-11-29</span></span></code></pre>
<pre><code><span><span class="co">## This is hpar version 1.41.1,</span></span>
<span><span class="co">## based on the Human Protein Atlas</span></span>
<span><span class="co">##   Version: 21.1</span></span>
<span><span class="co">##   Release data: 2022.05.31</span></span>
<span><span class="co">##   Ensembl build: 103.38</span></span>
<span><span class="co">## See '?hpar' or 'vignette('hpar')' for details.</span></span></code></pre>
<p>so that all the package’s functionality and data is available to the
user.</p>
</div>
</div>
<div class="section level2">
<h2 id="the-hpar-package">The <em><a href="https://bioconductor.org/packages/3.17/hpar" class="external-link">hpar</a></em>
package<a class="anchor" aria-label="anchor" href="#the-hpar-package"></a>
</h2>
<div class="section level3">
<h3 id="data-sets">Data sets<a class="anchor" aria-label="anchor" href="#data-sets"></a>
</h3>
<p>A table descibing all dataset available in the package can be
accessed with the <code><a href="../reference/getHpaDate.html">allHparData()</a></code> function.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hpa_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getHpaDate.html">allHparData</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div id="htmlwidget-13e3ac0af19385d95639" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-13e3ac0af19385d95639">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["hpaCancer16.1","hpaCancer","hpaNormalTissue16.1","hpaNormalTissue","hpaSecretome","hpaSubcellularLoc14","hpaSubcellularLoc16.1","hpaSubcellularLoc","rnaConsensusTissue","rnaFantomTissue","rnaGeneCellLine16.1","rnaGeneCellLine","rnaGeneTissue21.0","rnaGtexTissue","rnaHpaTissue"],["Pathology data: Staining profiles for proteins in human tumor tissue (version 16.1).","Pathology data: Staining profiles for proteins in human tumor tissue.","Normal tissue data: Expression profiles for proteins in human tissues based on immunohistochemisty using tissue micro arrays (version 16.1)","Normal tissue data: Expression profiles for proteins in human tissues based on immunohistochemisty using tissue micro arrays.","Secretome data: Ensembl genes with at least one predicted secreted transcript according to HPA predictions.","Subcellular location data: Subcellular location of proteins based on immunofluorescently stained cells (version 14).","Subcellular location data: Subcellular location of proteins based on immunofluorescently stained cells (version 16.1).","Subcellular location of proteins based on immunofluorescently stained cells.","RNA consensus tissue gene data: Consensus transcript expression levels summarized per gene in 54 tissues based on transcriptomics data from HPA and GTEx.","RNA FANTOM tissue gene data: Transcript expression levels summarized per gene in 60 tissues based on CAGE data","RNA HPA cell line gene data: Transcript expression levels summarized per gene in 69 cell lines (version 16.1).","RNA HPA cell line gene data: Transcript expression levels summarized per gene in 69 cell lines.","RNA HPA tissue gene data: Transcript expression levels summarized per gene in 37 tissues based on RNA-seq (version 21.0)","RNA GTEx tissue gene data: Transcript expression levels summarized per gene in 37 tissues based on RNA-seq","RNA HPA tissue gene data: Transcript expression levels summarized per gene in 256 tissues based on RNA-seq."],[3.16,3.16,3.16,3.16,3.16,3.16,3.16,3.16,3.16,3.16,3.16,3.16,3.16,3.16,3.16],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["TSV","TSV","TSV","TSV","TSV","TSV","TSV","TSV","TSV","TSV","TSV","TSV","TSV","TSV","TSV"],["https://www.proteinatlas.org/download","https://www.proteinatlas.org/download","https://www.proteinatlas.org/download","https://www.proteinatlas.org/download","https://www.proteinatlas.org/download","https://www.proteinatlas.org/download","https://www.proteinatlas.org/download","https://www.proteinatlas.org/download","https://www.proteinatlas.org/download","https://www.proteinatlas.org/download","https://www.proteinatlas.org/download","https://www.proteinatlas.org/download","https://www.proteinatlas.org/download","https://www.proteinatlas.org/download","https://www.proteinatlas.org/download"],[16.1,21,16.1,21,21,14,16.1,21,21,21,16.1,21,21,21,21],["Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens"],[9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606,9606],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],["Human Protein Atlas","Human Protein Atlas","Human Protein Atlas","Human Protein Atlas","Human Protein Atlas","Human Protein Atlas","Human Protein Atlas","Human Protein Atlas","Human Protein Atlas","Human Protein Atlas","Human Protein Atlas","Human Protein Atlas","Human Protein Atlas","Human Protein Atlas","Human Protein Atlas"],["Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;","Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;","Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;","Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;","Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;","Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;","Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;","Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;","Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;","Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;","Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;","Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;","Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;","Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;","Laurent Gatto &lt;laurent.gatto@uclouvain.be&gt;"],["data.frame","data.frame","data.frame","data.frame","data.frame","data.frame","data.frame","data.frame","data.frame","data.frame","data.frame","data.frame","data.frame","data.frame","data.frame"],["Rda","Rda","Rda","Rda","Rda","Rda","Rda","Rda","Rda","Rda","Rda","Rda","Rda","Rda","Rda"],["hpar/data/hpaCancer16.1.rda","hpar/data/hpaCancer.rda","hpar/data/hpaNormalTissue16.1.rda","hpar/data/hpaNormalTissue.rda","hpar/data/hpaSecretome.rda","hpar/data/hpaSubcellularLoc14.rda","hpar/data/hpaSubcellularLoc16.1.rda","hpar/data/hpaSubcellularLoc.rda","hpar/data/rnaConsensusTissue.rda","hpar/data/rnaFantomTissue.rda","hpar/data/rnaGeneCellLine16.1.rda","hpar/data/rnaGeneCellLine.rda","hpar/data/rnaGeneTissue21.0.rda","hpar/data/rnaGtexTissue.rda","hpar/data/rnaHpaTissue.rda"],["ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue","ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue","ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue","ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue","ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue","ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue","ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue","ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue","ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue","ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue","ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue","ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue","ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue","ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue","ExperimentHub:ExperimentData:ReproducibleResearch:RepositoryData:CancerData:Homo_sapiens_Data:Proteome:Tissue"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Title<\/th>\n      <th>Description<\/th>\n      <th>BiocVersion<\/th>\n      <th>Genome<\/th>\n      <th>SourceType<\/th>\n      <th>SourceUrl<\/th>\n      <th>SourceVersion<\/th>\n      <th>Species<\/th>\n      <th>TaxonomyId<\/th>\n      <th>Coordinate_1_based<\/th>\n      <th>DataProvider<\/th>\n      <th>Maintainer<\/th>\n      <th>RDataClass<\/th>\n      <th>DispatchClass<\/th>\n      <th>RDataPath<\/th>\n      <th>Tags<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,7,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><p>The <em>Title</em> variable corresponds to names of the data that can
be downloaded localled and cached as part of the ExperimentHub
infrastructure.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">normtissue</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hpar-package.html">hpaNormalTissue</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?hpar and browseVignettes('hpar') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">##              Gene Gene.name         Tissue           Cell.type        Level</span></span>
<span><span class="co">## 1 ENSG00000000003    TSPAN6 adipose tissue          adipocytes Not detected</span></span>
<span><span class="co">## 2 ENSG00000000003    TSPAN6  adrenal gland     glandular cells Not detected</span></span>
<span><span class="co">## 3 ENSG00000000003    TSPAN6       appendix     glandular cells       Medium</span></span>
<span><span class="co">## 4 ENSG00000000003    TSPAN6       appendix     lymphoid tissue Not detected</span></span>
<span><span class="co">## 5 ENSG00000000003    TSPAN6    bone marrow hematopoietic cells Not detected</span></span>
<span><span class="co">## 6 ENSG00000000003    TSPAN6         breast          adipocytes Not detected</span></span>
<span><span class="co">##   Reliability</span></span>
<span><span class="co">## 1    Approved</span></span>
<span><span class="co">## 2    Approved</span></span>
<span><span class="co">## 3    Approved</span></span>
<span><span class="co">## 4    Approved</span></span>
<span><span class="co">## 5    Approved</span></span>
<span><span class="co">## 6    Approved</span></span></code></pre>
<p>Note that given that the <code>hpa</code> data is distributed as par
the ExperimentHub infrastructure, it is also possible to query it
directly for relevant datasets.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"ExperimentHub"</span><span class="op">)</span></span>
<span><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">eh</span>, <span class="st">"hpar"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ExperimentHub with 15 records</span></span>
<span><span class="co">## # snapshotDate(): 2022-11-29</span></span>
<span><span class="co">## # $dataprovider: Human Protein Atlas</span></span>
<span><span class="co">## # $species: Homo sapiens</span></span>
<span><span class="co">## # $rdataclass: data.frame</span></span>
<span><span class="co">## # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">## #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">## #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">## # retrieve records with, e.g., 'object[["EH7765"]]' </span></span>
<span><span class="co">## </span></span>
<span><span class="co">##            title              </span></span>
<span><span class="co">##   EH7765 | hpaCancer16.1      </span></span>
<span><span class="co">##   EH7766 | hpaCancer          </span></span>
<span><span class="co">##   EH7767 | hpaNormalTissue16.1</span></span>
<span><span class="co">##   EH7768 | hpaNormalTissue    </span></span>
<span><span class="co">##   EH7769 | hpaSecretome       </span></span>
<span><span class="co">##   ...      ...                </span></span>
<span><span class="co">##   EH7775 | rnaGeneCellLine16.1</span></span>
<span><span class="co">##   EH7776 | rnaGeneCellLine    </span></span>
<span><span class="co">##   EH7777 | rnaGeneTissue21.0  </span></span>
<span><span class="co">##   EH7778 | rnaGtexTissue      </span></span>
<span><span class="co">##   EH7779 | rnaHpaTissue</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="hpa-interface">HPA interface<a class="anchor" aria-label="anchor" href="#hpa-interface"></a>
</h3>
<p>Each data described above is a <code>data.frame</code> and can be
easily manipulated using standard R or
<code>BiocStyle::CRANpkg("tidyverse")</code> tidyverse
functionality.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">normtissue</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Gene"        "Gene.name"   "Tissue"      "Cell.type"   "Level"      </span></span>
<span><span class="co">## [6] "Reliability"</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Number of genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">normtissue</span><span class="op">$</span><span class="va">Gene</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 15323</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Number of cell types</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">normtissue</span><span class="op">$</span><span class="va">Cell.type</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 141</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Number of tissues</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">normtissue</span><span class="op">$</span><span class="va">Tissue</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 63</span></span></code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Number of genes highlighly and reliably expressed in each cell type</span></span>
<span><span class="co">## in each tissue.</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">"dplyr"</a></span><span class="op">)</span></span>
<span><span class="va">normtissue</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Reliability</span> <span class="op">==</span> <span class="st">"Approved"</span>,</span>
<span>           <span class="va">Level</span> <span class="op">==</span> <span class="st">"High"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">Cell.type</span>, <span class="va">Tissue</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          Cell.type          Tissue    n</span></span>
<span><span class="co">## 1  glandular cells     gallbladder 1578</span></span>
<span><span class="co">## 2  glandular cells        duodenum 1551</span></span>
<span><span class="co">## 3  glandular cells small intestine 1532</span></span>
<span><span class="co">## 4  glandular cells          rectum 1490</span></span>
<span><span class="co">## 5  glandular cells           colon 1404</span></span>
<span><span class="co">## 6 cells in tubules          kidney 1357</span></span></code></pre>
<p>We will illustrate additional datasets using the TSPAN6 (tetraspanin
6) gene (ENSG00000000003) as example.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">id</span> <span class="op">&lt;-</span> <span class="st">"ENSG00000000003"</span></span>
<span><span class="va">subcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hpar-package.html">hpaSubcellularLoc</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?hpar and browseVignettes('hpar') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rna</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hpar-package.html">rnaGeneCellLine</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?hpar and browseVignettes('hpar') for documentation</span></span>
<span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Compine protein immunohistochemisty data, with the subcellular</span></span>
<span><span class="co">## location and RNA expression levels.</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">normtissue</span>, <span class="va">Gene</span> <span class="op">==</span> <span class="va">id</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">full_join</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">subcell</span>, <span class="va">Gene</span> <span class="op">==</span> <span class="va">id</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">full_join</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">rna</span>, <span class="va">Gene</span> <span class="op">==</span> <span class="va">id</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Joining, by = c("Gene", "Gene.name", "Reliability")</span></span></code></pre>
<pre><code><span><span class="co">## Joining, by = c("Gene", "Gene.name")</span></span></code></pre>
<pre><code><span><span class="co">##              Gene Gene.name         Tissue  Cell.type        Level Reliability</span></span>
<span><span class="co">## 1 ENSG00000000003    TSPAN6 adipose tissue adipocytes Not detected    Approved</span></span>
<span><span class="co">## 2 ENSG00000000003    TSPAN6 adipose tissue adipocytes Not detected    Approved</span></span>
<span><span class="co">## 3 ENSG00000000003    TSPAN6 adipose tissue adipocytes Not detected    Approved</span></span>
<span><span class="co">## 4 ENSG00000000003    TSPAN6 adipose tissue adipocytes Not detected    Approved</span></span>
<span><span class="co">## 5 ENSG00000000003    TSPAN6 adipose tissue adipocytes Not detected    Approved</span></span>
<span><span class="co">## 6 ENSG00000000003    TSPAN6 adipose tissue adipocytes Not detected    Approved</span></span>
<span><span class="co">##            Main.location       Additional.location Extracellular.location</span></span>
<span><span class="co">## 1 Cell Junctions;Cytosol Nucleoli fibrillar center                       </span></span>
<span><span class="co">## 2 Cell Junctions;Cytosol Nucleoli fibrillar center                       </span></span>
<span><span class="co">## 3 Cell Junctions;Cytosol Nucleoli fibrillar center                       </span></span>
<span><span class="co">## 4 Cell Junctions;Cytosol Nucleoli fibrillar center                       </span></span>
<span><span class="co">## 5 Cell Junctions;Cytosol Nucleoli fibrillar center                       </span></span>
<span><span class="co">## 6 Cell Junctions;Cytosol Nucleoli fibrillar center                       </span></span>
<span><span class="co">##   Enhanced Supported                                         Approved Uncertain</span></span>
<span><span class="co">## 1                    Cell Junctions;Cytosol;Nucleoli fibrillar center          </span></span>
<span><span class="co">## 2                    Cell Junctions;Cytosol;Nucleoli fibrillar center          </span></span>
<span><span class="co">## 3                    Cell Junctions;Cytosol;Nucleoli fibrillar center          </span></span>
<span><span class="co">## 4                    Cell Junctions;Cytosol;Nucleoli fibrillar center          </span></span>
<span><span class="co">## 5                    Cell Junctions;Cytosol;Nucleoli fibrillar center          </span></span>
<span><span class="co">## 6                    Cell Junctions;Cytosol;Nucleoli fibrillar center          </span></span>
<span><span class="co">##   Single.cell.variation.intensity Single.cell.variation.spatial</span></span>
<span><span class="co">## 1                         Cytosol                              </span></span>
<span><span class="co">## 2                         Cytosol                              </span></span>
<span><span class="co">## 3                         Cytosol                              </span></span>
<span><span class="co">## 4                         Cytosol                              </span></span>
<span><span class="co">## 5                         Cytosol                              </span></span>
<span><span class="co">## 6                         Cytosol                              </span></span>
<span><span class="co">##   Cell.cycle.dependency</span></span>
<span><span class="co">## 1                      </span></span>
<span><span class="co">## 2                      </span></span>
<span><span class="co">## 3                      </span></span>
<span><span class="co">## 4                      </span></span>
<span><span class="co">## 5                      </span></span>
<span><span class="co">## 6                      </span></span>
<span><span class="co">##                                                                                     GO.id</span></span>
<span><span class="co">## 1 Cell Junctions (GO:0030054);Cytosol (GO:0005829);Nucleoli fibrillar center (GO:0001650)</span></span>
<span><span class="co">## 2 Cell Junctions (GO:0030054);Cytosol (GO:0005829);Nucleoli fibrillar center (GO:0001650)</span></span>
<span><span class="co">## 3 Cell Junctions (GO:0030054);Cytosol (GO:0005829);Nucleoli fibrillar center (GO:0001650)</span></span>
<span><span class="co">## 4 Cell Junctions (GO:0030054);Cytosol (GO:0005829);Nucleoli fibrillar center (GO:0001650)</span></span>
<span><span class="co">## 5 Cell Junctions (GO:0030054);Cytosol (GO:0005829);Nucleoli fibrillar center (GO:0001650)</span></span>
<span><span class="co">## 6 Cell Junctions (GO:0030054);Cytosol (GO:0005829);Nucleoli fibrillar center (GO:0001650)</span></span>
<span><span class="co">##   Cell.line  TPM pTPM nTPM</span></span>
<span><span class="co">## 1     A-431 21.3 25.6 24.8</span></span>
<span><span class="co">## 2      A549 20.5 24.4 23.0</span></span>
<span><span class="co">## 3      AF22 78.8 96.9 80.6</span></span>
<span><span class="co">## 4    AN3-CA 38.7 47.1 42.2</span></span>
<span><span class="co">## 5  ASC diff 19.2 22.1 28.7</span></span>
<span><span class="co">## 6 ASC TERT1 11.3 13.1 16.6</span></span></code></pre>
<p>It is also possible to directly open the HPA page for a specific gene
(see figure below).</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/browseHPA.html">browseHPA</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="hpa.png" alt=""><p class="caption">The HPA web page for the tetraspanin 6 gene
(ENSG00000000003).</p>
</div>
</div>
<div class="section level3">
<h3 id="hpa-release-information">HPA release information<a class="anchor" aria-label="anchor" href="#hpa-release-information"></a>
</h3>
<p>Information about the HPA release used to build the installed <em><a href="https://bioconductor.org/packages/3.17/hpar" class="external-link">hpar</a></em> package
can be accessed with <code>getHpaVersion</code>, <code>getHpaDate</code>
and <code>getHpaEnsembl</code>. Full release details can be found on the
<a href="http://www.proteinatlas.org/about/releases" class="external-link">HPA release
history</a> page.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getHpaDate.html">getHpaVersion</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## version </span></span>
<span><span class="co">##  "21.1"</span></span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getHpaDate.html">getHpaDate</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         date </span></span>
<span><span class="co">## "2022.05.31"</span></span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getHpaDate.html">getHpaEnsembl</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  ensembl </span></span>
<span><span class="co">## "103.38"</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="a-small-use-case">A small use case<a class="anchor" aria-label="anchor" href="#a-small-use-case"></a>
</h2>
<p>Let’s compare the subcellular localisation annotation obtained from
the HPA subcellular location data set and the information available in
the Bioconductor annotation packages.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">id</span> <span class="op">&lt;-</span> <span class="st">"ENSG00000001460"</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">subcell</span>, <span class="va">Gene</span> <span class="op">==</span> <span class="va">id</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##              Gene Gene.name Reliability Main.location Additional.location</span></span>
<span><span class="co">## 1 ENSG00000001460     STPG1    Approved   Nucleoplasm                    </span></span>
<span><span class="co">##   Extracellular.location Enhanced Supported    Approved Uncertain</span></span>
<span><span class="co">## 1                                           Nucleoplasm          </span></span>
<span><span class="co">##   Single.cell.variation.intensity Single.cell.variation.spatial</span></span>
<span><span class="co">## 1                                                              </span></span>
<span><span class="co">##   Cell.cycle.dependency                    GO.id</span></span>
<span><span class="co">## 1                       Nucleoplasm (GO:0005654)</span></span></code></pre>
<p>Below, we first extract all cellular component GO terms available for
<code>id</code> from the <em><a href="https://bioconductor.org/packages/3.17/org.Hs.eg.db" class="external-link">org.Hs.eg.db</a></em>
human annotation and then retrieve their term definitions using the
<em><a href="https://bioconductor.org/packages/3.17/GO.db" class="external-link">GO.db</a></em>
database.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"GO.db"</span><span class="op">)</span></span>
<span><span class="va">ans</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keys <span class="op">=</span> <span class="va">id</span>,</span>
<span>                             columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSEMBL"</span>, <span class="st">"GO"</span>, <span class="st">"ONTOLOGY"</span><span class="op">)</span>,</span>
<span>                             keytype <span class="op">=</span> <span class="st">"ENSEMBL"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ans</span> <span class="op">&lt;-</span> <span class="va">ans</span><span class="op">[</span><span class="va">ans</span><span class="op">$</span><span class="va">ONTOLOGY</span> <span class="op">==</span> <span class="st">"CC"</span>, <span class="op">]</span></span>
<span><span class="va">ans</span></span></code></pre></div>
<pre><code><span><span class="co">##           ENSEMBL         GO EVIDENCE ONTOLOGY</span></span>
<span><span class="co">## 2 ENSG00000001460 GO:0005634      IEA       CC</span></span>
<span><span class="co">## 3 ENSG00000001460 GO:0005739      IEA       CC</span></span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">GOTERM</span><span class="op">[</span><span class="va">ans</span><span class="op">$</span><span class="va">GO</span><span class="op">]</span><span class="op">)</span>, <span class="va">slot</span>, <span class="st">"Term"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      GO:0005634      GO:0005739 </span></span>
<span><span class="co">##       "nucleus" "mitochondrion"</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">## R Under development (unstable) (2022-11-23 r83380)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] ExperimentHub_2.7.0  AnnotationHub_3.7.0  BiocFileCache_2.7.0 </span></span>
<span><span class="co">##  [4] dbplyr_2.2.1         hpar_1.41.1          GO.db_3.16.0        </span></span>
<span><span class="co">##  [7] org.Hs.eg.db_3.16.0  AnnotationDbi_1.61.0 IRanges_2.33.0      </span></span>
<span><span class="co">## [10] S4Vectors_0.37.0     Biobase_2.59.0       BiocGenerics_0.45.0 </span></span>
<span><span class="co">## [13] dplyr_1.0.10         BiocStyle_2.27.0    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] tidyselect_1.2.0              blob_1.2.3                   </span></span>
<span><span class="co">##  [3] filelock_1.0.2                Biostrings_2.67.0            </span></span>
<span><span class="co">##  [5] bitops_1.0-7                  fastmap_1.1.0                </span></span>
<span><span class="co">##  [7] RCurl_1.98-1.9                promises_1.2.0.1             </span></span>
<span><span class="co">##  [9] digest_0.6.30                 mime_0.12                    </span></span>
<span><span class="co">## [11] lifecycle_1.0.3               ellipsis_0.3.2               </span></span>
<span><span class="co">## [13] KEGGREST_1.39.0               interactiveDisplayBase_1.37.0</span></span>
<span><span class="co">## [15] RSQLite_2.2.19                magrittr_2.0.3               </span></span>
<span><span class="co">## [17] compiler_4.3.0                rlang_1.0.6                  </span></span>
<span><span class="co">## [19] sass_0.4.4                    tools_4.3.0                  </span></span>
<span><span class="co">## [21] utf8_1.2.2                    yaml_2.3.6                   </span></span>
<span><span class="co">## [23] knitr_1.41                    htmlwidgets_1.5.4            </span></span>
<span><span class="co">## [25] bit_4.0.5                     curl_4.3.3                   </span></span>
<span><span class="co">## [27] withr_2.5.0                   purrr_0.3.5                  </span></span>
<span><span class="co">## [29] desc_1.4.2                    fansi_1.0.3                  </span></span>
<span><span class="co">## [31] xtable_1.8-4                  cli_3.4.1                    </span></span>
<span><span class="co">## [33] rmarkdown_2.18                crayon_1.5.2                 </span></span>
<span><span class="co">## [35] ragg_1.2.4                    generics_0.1.3               </span></span>
<span><span class="co">## [37] httr_1.4.4                    DBI_1.1.3                    </span></span>
<span><span class="co">## [39] cachem_1.0.6                  stringr_1.4.1                </span></span>
<span><span class="co">## [41] zlibbioc_1.45.0               assertthat_0.2.1             </span></span>
<span><span class="co">## [43] BiocManager_1.30.19           XVector_0.39.0               </span></span>
<span><span class="co">## [45] vctrs_0.5.1                   jsonlite_1.8.3               </span></span>
<span><span class="co">## [47] bookdown_0.30                 bit64_4.0.5                  </span></span>
<span><span class="co">## [49] systemfonts_1.0.4             crosstalk_1.2.0              </span></span>
<span><span class="co">## [51] jquerylib_0.1.4               glue_1.6.2                   </span></span>
<span><span class="co">## [53] pkgdown_2.0.6.9000            DT_0.26                      </span></span>
<span><span class="co">## [55] stringi_1.7.8                 BiocVersion_3.17.1           </span></span>
<span><span class="co">## [57] later_1.3.0                   GenomeInfoDb_1.35.5          </span></span>
<span><span class="co">## [59] tibble_3.1.8                  pillar_1.8.1                 </span></span>
<span><span class="co">## [61] rappdirs_0.3.3                htmltools_0.5.3              </span></span>
<span><span class="co">## [63] GenomeInfoDbData_1.2.9        R6_2.5.1                     </span></span>
<span><span class="co">## [65] textshaping_0.3.6             rprojroot_2.0.3              </span></span>
<span><span class="co">## [67] evaluate_0.18                 shiny_1.7.3                  </span></span>
<span><span class="co">## [69] png_0.1-8                     memoise_2.0.1                </span></span>
<span><span class="co">## [71] httpuv_1.6.6                  bslib_0.4.1                  </span></span>
<span><span class="co">## [73] Rcpp_1.0.9                    xfun_0.35                    </span></span>
<span><span class="co">## [75] fs_1.5.2                      pkgconfig_2.0.3</span></span></code></pre>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Uhlen2010" class="csl-entry">
Uhlen, Mathias, Per Oksvold, Linn Fagerberg, Emma Lundberg, Kalle
Jonasson, Mattias Forsberg, Martin Zwahlen, et al. 2010. <span>“<span class="nocase">Towards a knowledge-based Human Protein
Atlas.</span>”</span> <em>Nature Biotechnology</em> 28 (12): 1248–50. <a href="https://doi.org/10.1038/nbt1210-1248" class="external-link">https://doi.org/10.1038/nbt1210-1248</a>.
</div>
<div id="ref-Uhlen2005" class="csl-entry">
Uhlén, Mathias, Erik Björling, Charlotta Agaton, Cristina Al-Khalili A.
Szigyarto, Bahram Amini, Elisabet Andersen, Ann-Catrin C. Andersson, et
al. 2005. <span>“<span class="nocase">A human protein atlas for normal
and cancer tissues based on antibody proteomics.</span>”</span>
<em>Molecular &amp; Cellular Proteomics : MCP</em> 4 (12): 1920–32. <a href="https://doi.org/10.1074/mcp.M500279-MCP200" class="external-link">https://doi.org/10.1074/mcp.M500279-MCP200</a>.
</div>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p>www.proteinatlas.org<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Laurent Gatto, Manon Martin.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
