<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Human Protein Atlas in R • hpar</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Human Protein Atlas in R">
<meta property="og:description" content="hpar">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">hpar</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.33.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/hpar.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="hpar_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Human Protein Atlas in R</h1>
                        <h4 class="author">Laurent Gatto</h4>
            <address class="author_afil">
      de Duve Institute, UCLouvain, Belgium<br><div class="hidden name"><code>hpar.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The Human Protein Atlas (HPA) is a systematic study oh the human proteome using antibody-based proteomics. Multiple tissues and cell lines are systematically assayed affinity-purified antibodies and confocal microscopy. The <em>hpar</em> package is an R interface to the HPA project. It distributes three data sets, provides functionality to query these and to access detailed information pages, including confocal microscopy images available on the HPA web page.</p>
    </div>
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<div id="the-hpa-project" class="section level2">
<h2 class="hasAnchor">
<a href="#the-hpa-project" class="anchor"></a>The HPA project</h2>
<p>From the <a href="http://www.proteinatlas.org/">Human Protein Atlas</a> <span class="citation">(Uhlén et al. 2005; Uhlen et al. 2010)</span> site:</p>
<blockquote>
<p>The Swedish Human Protein Atlas project, funded by the Knut and Alice Wallenberg Foundation, has been set up to allow for a systematic exploration of the human proteome using Antibody-Based Proteomics. This is accomplished by combining high-throughput generation of affinity-purified antibodies with protein profiling in a multitude of tissues and cells assembled in tissue microarrays. Confocal microscopy analysis using human cell lines is performed for more detailed protein localisation. The program hosts the Human Protein Atlas portal with expression profiles of human proteins in tissues and cells.</p>
</blockquote>
<p>The <em><a href="https://bioconductor.org/packages/3.13/hpar">hpar</a></em> package provides access to HPA data from the R interface. It also distributes the following data sets:</p>
<ul>
<li><p><code>hpaNormalTissue</code> <em>Normal tissue data</em>: Expression profiles for proteins in human tissues based on immunohistochemisty using tissue micro arrays. The tab-separated file includes Ensembl gene identifier (“Gene”), tissue name (“Tissue”), annotated cell type (“Cell type”), expression value (“Level”), and the gene reliability of the expression value (“Reliability”).}</p></li>
<li><p><code>hpaNormalTissue16.1</code>: Same as above, for version 16.1.</p></li>
<li><p><code>hpaCancer</code> <em>Pathology data</em>: Staining profiles for proteins in human tumor tissue based on immunohistochemisty using tissue micro arrays and log-rank P value for Kaplan-Meier analysis of correlation between mRNA expression level and patient survival. The tab-separated file includes Ensembl gene identifier (“Gene”), gene name (“Gene name”), tumor name (“Cancer”), the number of patients annotated for different staining levels (“High”, “Medium”, “Low” &amp; “Not detected”) and log-rank p values for patient survival and mRNA correlation (“prognostic - favourable”, “unprognostic - favourable”, “prognostic - unfavourable”, “unprognostic - unfavourable”). }</p></li>
<li><p><code>hpaCancer16.1</code>: Same as above, for version 16.1.</p></li>
<li><p><code>rnaGeneTissue</code> <em>RNA HPA tissue gene data</em>: Transcript expression levels summarized per gene in 37 tissues based on RNA-seq. The tab-separated file includes Ensembl gene identifier (“Gene”), analysed sample (“Tissue”), transcripts per million (“TPM”), protein-transcripts per million (“pTPM”) and normalized expression (“NX”). }</p></li>
<li><p><code>rnaGeneCellLine</code> <em>RNA HPA cell line gene data</em>: Transcript expression levels summarized per gene in 64 cell lines. The tab-separated file includes Ensembl gene identifier (“Gene”), analysed sample (“Cell line”), transcripts per million (“TPM”), protein-coding transcripts per million (“pTPM”) and normalized expression (“NX”). }</p></li>
<li><p><code>rnaGeneCellLine16.1</code>: Same as above, for version 16.1.</p></li>
<li><p><code>hpaSubcellularLoc</code> <em>Subcellular location data</em>: Subcellular location of proteins based on immunofluorescently stained cells. The tab-separated file includes the following columns: Ensembl gene identifier (“Gene”), name of gene (“Gene name”), gene reliability score (“Reliability”), enhanced locations (“Enhanced”), supported locations (“Supported”), Approved locations (“Approved”), uncertain locations (“Uncertain”), locations with single-cell variation in intensity (“Single-cell variation intensity”), locations with spatial single-cell variation (“Single-cell variation spatial”), locations with observed cell cycle dependency (type can be one or more of biological definition, custom data or correlation) (“Cell cycle dependency”), Gene Ontology Cellular Component term identifier (“GO id”).}</p></li>
<li><p><code>hpaSubcellularLoc14</code> and <code>*16.1</code>: Same as above, for versions 14 and 16.1.</p></li>
<li><p><code>hpaSecretome</code> <em>Secretome data</em>: The human secretome is here defined as all Ensembl genes with at least one predicted secreted transcript according to HPA predictions. The complete information about the HPA Secretomedata is given on . This dataset has 230 columns and includes the Ensembl gene identifier (“Gene”). Information about the additionnal variables can be found  by clicking on .</p></li>
</ul>
</div>
<div id="hpa-data-usage-policy" class="section level2">
<h2 class="hasAnchor">
<a href="#hpa-data-usage-policy" class="anchor"></a>HPA data usage policy</h2>
<p>The use of data and images from the HPA in publications and presentations is permitted provided that the following conditions are met:</p>
<ul>
<li>The publication and/or presentation are solely for informational and non-commercial purposes.</li>
<li>The source of the data and/or image is referred to the HPA site<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> and/or one or more of our publications are cited.</li>
</ul>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p><em><a href="https://bioconductor.org/packages/3.13/hpar">hpar</a></em> is available through the Bioconductor project. Details about the package and the installation procedure can be found on its <a href="http://bioconductor.org/packages/devel/bioc/html/hpar.html">landing page</a>. To install using the dedicated Bioconductor infrastructure, run :</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## install BiocManager only one</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="co">## install hpar</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"hpar"</span><span class="op">)</span></code></pre></div>
<p>After installation, <em><a href="https://bioconductor.org/packages/3.13/hpar">hpar</a></em> will have to be explicitly loaded with</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"hpar"</span><span class="op">)</span></code></pre></div>
<pre><code>## This is hpar version 1.33.2,
## based on the Human Protein Atlas
##   Version: 20.0
##   Release data: 2020.11.19
##   Ensembl build: 92.38
## See '?hpar' or 'vignette('hpar')' for details.</code></pre>
<p>so that all the package’s functionality and data is available to the user.</p>
</div>
</div>
<div id="the-hpar-package" class="section level1">
<h1 class="hasAnchor">
<a href="#the-hpar-package" class="anchor"></a>The <em><a href="https://bioconductor.org/packages/3.13/hpar">hpar</a></em> package</h1>
<div id="data-sets" class="section level2">
<h2 class="hasAnchor">
<a href="#data-sets" class="anchor"></a>Data sets</h2>
<p>The data sets described above can be loaded with the <code>data</code> function, as illustrated below for <code>hpaNormalTissue</code> below. Each data set is a <code>data.frame</code> and can be easily manipulated using standard R functionality. The code chunk below illustrates some of its properties.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">hpaNormalTissue</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">hpaNormalTissue</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 1118517       6</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">hpaNormalTissue</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "Gene"        "Gene.name"   "Tissue"      "Cell.type"   "Level"      
## [6] "Reliability"</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Number of genes</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">hpaNormalTissue</span><span class="op">$</span><span class="va">Gene</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 15320</code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Number of cell types</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">hpaNormalTissue</span><span class="op">$</span><span class="va">Cell.type</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 120</code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">hpaNormalTissue</span><span class="op">$</span><span class="va">Cell.type</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Number of tissues</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">hpaNormalTissue</span><span class="op">$</span><span class="va">Tissue</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 63</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">hpaNormalTissue</span><span class="op">$</span><span class="va">Tissue</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## NULL</code></pre>
</div>
<div id="hpa-interface" class="section level2">
<h2 class="hasAnchor">
<a href="#hpa-interface" class="anchor"></a>HPA interface</h2>
<p>The package provides a interface to the HPA data. The <code>getHpa</code> allows to query the data sets described above. It takes three arguments, <code>id</code>, <code>hpadata</code> and <code>type</code>, that control the query, what data set to interrogate and how to report results respectively. The HPA data uses Ensembl gene identifiers and <code>id</code> must be a valid identifier. <code>hpadata</code> must be one of available dataset. <code>type</code> can be either <code>"data"</code> or <code>"details"</code>. The former is the default and returns a <code>data.frame</code> containing the information relevant to <code>id</code>. It is also possible to obtained detailed information, (including cell images) as web pages, directly from the HPA web page, using <code>"details"</code>.</p>
<p>We will illustrate this functionality with using the TSPAN6 (tetraspanin 6) gene (ENSG00000000003) as example.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">id</span> <span class="op">&lt;-</span> <span class="st">"ENSG00000000003"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/getHpa.html">getHpa</a></span><span class="op">(</span><span class="va">id</span>, hpadata <span class="op">=</span> <span class="st">"hpaNormalTissue"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##              Gene Gene.name         Tissue           Cell.type        Level
## 1 ENSG00000000003    TSPAN6 adipose tissue          adipocytes Not detected
## 2 ENSG00000000003    TSPAN6  adrenal gland     glandular cells Not detected
## 3 ENSG00000000003    TSPAN6       appendix     glandular cells       Medium
## 4 ENSG00000000003    TSPAN6       appendix     lymphoid tissue Not detected
## 5 ENSG00000000003    TSPAN6    bone marrow hematopoietic cells Not detected
## 6 ENSG00000000003    TSPAN6         breast          adipocytes Not detected
##   Reliability
## 1    Approved
## 2    Approved
## 3    Approved
## 4    Approved
## 5    Approved
## 6    Approved</code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getHpa.html">getHpa</a></span><span class="op">(</span><span class="va">id</span>, hpadata <span class="op">=</span> <span class="st">"hpaSubcellularLoc"</span><span class="op">)</span></code></pre></div>
<pre><code>##              Gene Gene.name Reliability          Main.location
## 1 ENSG00000000003    TSPAN6    Approved Cell Junctions;Cytosol
##         Additional.location Extracellular.location Enhanced Supported
## 1 Nucleoli fibrillar center                                          
##                                           Approved Uncertain
## 1 Cell Junctions;Cytosol;Nucleoli fibrillar center          
##   Single.cell.variation.intensity Single.cell.variation.spatial
## 1                         Cytosol                              
##   Cell.cycle.dependency
## 1                      
##                                                                                     GO.id
## 1 Cell Junctions (GO:0030054);Cytosol (GO:0005829);Nucleoli fibrillar center (GO:0001650)</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/getHpa.html">getHpa</a></span><span class="op">(</span><span class="va">id</span>, hpadata <span class="op">=</span> <span class="st">"rnaGeneCellLine"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##              Gene Gene.name Cell.line   TPM  pTPM   NX
## 1 ENSG00000000003    TSPAN6     A-431  27.8  33.9  7.9
## 2 ENSG00000000003    TSPAN6      A549  37.6  45.5 10.6
## 3 ENSG00000000003    TSPAN6      AF22 108.1 134.5 28.7
## 4 ENSG00000000003    TSPAN6    AN3-CA  51.8  64.4 14.5
## 5 ENSG00000000003    TSPAN6  ASC diff  32.3  37.4 12.6
## 6 ENSG00000000003    TSPAN6 ASC TERT1  17.7  20.8  6.8</code></pre>
<p>If we ask for <code>"detail"</code>, a browser page pointing to the relevant page is open (see figure below)</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getHpa.html">getHpa</a></span><span class="op">(</span><span class="va">id</span>, type <span class="op">=</span> <span class="st">"details"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="hpa.png" alt=""><p class="caption">The HPA web page for the tetraspanin 6 gene (ENSG00000000003).</p>
</div>
<p>If a user is interested specifically in one data set, it is possible to set <code>hpadata</code> globally and omit it in <code>getHpa</code>. This is done by setting the <code>hpar</code> options <code>hpardata</code> with the <code>setHparOptions</code> function. The current default data set can be tested with <code>getHparOptions</code>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getHparOptions.html">getHparOptions</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## $hpar
## $hpar$hpadata
## [1] "hpaNormalTissue"</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/setHparOptions.html">setHparOptions</a></span><span class="op">(</span>hpadata <span class="op">=</span> <span class="st">"hpaSubcellularLoc"</span><span class="op">)</span>
<span class="fu"><a href="../reference/getHparOptions.html">getHparOptions</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## $hpar
## $hpar$hpadata
## [1] "hpaSubcellularLoc"</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getHpa.html">getHpa</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span></code></pre></div>
<pre><code>##              Gene Gene.name Reliability          Main.location
## 1 ENSG00000000003    TSPAN6    Approved Cell Junctions;Cytosol
##         Additional.location Extracellular.location Enhanced Supported
## 1 Nucleoli fibrillar center                                          
##                                           Approved Uncertain
## 1 Cell Junctions;Cytosol;Nucleoli fibrillar center          
##   Single.cell.variation.intensity Single.cell.variation.spatial
## 1                         Cytosol                              
##   Cell.cycle.dependency
## 1                      
##                                                                                     GO.id
## 1 Cell Junctions (GO:0030054);Cytosol (GO:0005829);Nucleoli fibrillar center (GO:0001650)</code></pre>
</div>
<div id="hpa-release-information" class="section level2">
<h2 class="hasAnchor">
<a href="#hpa-release-information" class="anchor"></a>HPA release information</h2>
<p>Information about the HPA release used to build the installed</p>
<p><em><a href="https://bioconductor.org/packages/3.13/hpar">hpar</a></em> package can be accessed with <code>getHpaVersion</code>, <code>getHpaDate</code> and <code>getHpaEnsembl</code>. Full release details can be found on the <a href="http://www.proteinatlas.org/about/releases">HPA release history</a> page.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getHpaDate.html">getHpaVersion</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## version 
##  "20.0"</code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getHpaDate.html">getHpaDate</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>##         date 
## "2020.11.19"</code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getHpaDate.html">getHpaEnsembl</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## ensembl 
## "92.38"</code></pre>
</div>
</div>
<div id="a-small-use-case" class="section level1">
<h1 class="hasAnchor">
<a href="#a-small-use-case" class="anchor"></a>A small use case</h1>
<p>Let’s compare the subcellular localisation annotation obtained from the HPA subcellular location data set and the information available in the Bioconductor annotation packages.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">id</span> <span class="op">&lt;-</span> <span class="st">"ENSG00000001460"</span>
<span class="fu"><a href="../reference/getHpa.html">getHpa</a></span><span class="op">(</span><span class="va">id</span>, <span class="st">"hpaSubcellularLoc"</span><span class="op">)</span></code></pre></div>
<pre><code>##              Gene Gene.name Reliability Main.location Additional.location
## 8 ENSG00000001460     STPG1    Approved   Nucleoplasm                    
##   Extracellular.location Enhanced Supported    Approved Uncertain
## 8                                           Nucleoplasm          
##   Single.cell.variation.intensity Single.cell.variation.spatial
## 8                                                              
##   Cell.cycle.dependency                    GO.id
## 8                       Nucleoplasm (GO:0005654)</code></pre>
<p>Below, we first extract all cellular component GO terms available for <code>id</code> from the <em><a href="https://bioconductor.org/packages/3.13/org.Hs.eg.db">org.Hs.eg.db</a></em> human annotation and then retrieve their term definitions using the <em><a href="https://bioconductor.org/packages/3.13/GO.db">GO.db</a></em> database.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"GO.db"</span><span class="op">)</span>
<span class="va">ans</span> <span class="op">&lt;-</span> <span class="fu">select</span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keys <span class="op">=</span> <span class="va">id</span>,
              columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ENSEMBL"</span>, <span class="st">"GO"</span>, <span class="st">"ONTOLOGY"</span><span class="op">)</span>,
              keytype <span class="op">=</span> <span class="st">"ENSEMBL"</span><span class="op">)</span></code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ans</span> <span class="op">&lt;-</span> <span class="va">ans</span><span class="op">[</span><span class="va">ans</span><span class="op">$</span><span class="va">ONTOLOGY</span> <span class="op">==</span> <span class="st">"CC"</span>, <span class="op">]</span>
<span class="va">ans</span></code></pre></div>
<pre><code>##           ENSEMBL         GO EVIDENCE ONTOLOGY
## 2 ENSG00000001460 GO:0005634      IEA       CC
## 3 ENSG00000001460 GO:0005739      IEA       CC</code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span><span class="op">(</span><span class="va">GOTERM</span><span class="op">[</span><span class="va">ans</span><span class="op">$</span><span class="va">GO</span><span class="op">]</span><span class="op">)</span>, <span class="va">slot</span>, <span class="st">"Term"</span><span class="op">)</span></code></pre></div>
<pre><code>##      GO:0005634      GO:0005739 
##       "nucleus" "mitochondrion"</code></pre>
</div>
<div id="session-information" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<pre><code>## R Under development (unstable) (2020-11-18 r79449)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.1 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
## [1] hpar_1.33.2          GO.db_3.12.1         org.Hs.eg.db_3.12.0 
## [4] AnnotationDbi_1.53.0 IRanges_2.25.2       S4Vectors_0.29.3    
## [7] Biobase_2.51.0       BiocGenerics_0.37.0  BiocStyle_2.19.1    
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.5          compiler_4.1.0      BiocManager_1.30.10
##  [4] tools_4.1.0         digest_0.6.27       bit_4.0.4          
##  [7] evaluate_0.14       memoise_1.1.0       RSQLite_2.2.1      
## [10] pkgconfig_2.0.3     rlang_0.4.9         DBI_1.1.0          
## [13] yaml_2.2.1          pkgdown_1.6.1.9000  xfun_0.19          
## [16] stringr_1.4.0       knitr_1.30          desc_1.2.0         
## [19] fs_1.5.0            vctrs_0.3.5         systemfonts_0.3.2  
## [22] rprojroot_2.0.2     bit64_4.0.5         R6_2.5.0           
## [25] textshaping_0.2.1   rmarkdown_2.5       bookdown_0.21      
## [28] blob_1.2.1          magrittr_2.0.1      htmltools_0.5.0    
## [31] assertthat_0.2.1    ragg_0.4.0          stringi_1.5.3      
## [34] crayon_1.3.4</code></pre>
<div id="refs" class="references">
<div id="ref-Uhlen2010">
<p>Uhlen, Mathias, Per Oksvold, Linn Fagerberg, Emma Lundberg, Kalle Jonasson, Mattias Forsberg, Martin Zwahlen, et al. 2010. “Towards a knowledge-based Human Protein Atlas.” <em>Nature Biotechnology</em> 28 (12): 1248–50. <a href="https://doi.org/10.1038/nbt1210-1248">https://doi.org/10.1038/nbt1210-1248</a>.</p>
</div>
<div id="ref-Uhlen2005">
<p>Uhlén, Mathias, Erik Björling, Charlotta Agaton, Cristina Al-Khalili A. Szigyarto, Bahram Amini, Elisabet Andersen, Ann-Catrin C. Andersson, et al. 2005. “A human protein atlas for normal and cancer tissues based on antibody proteomics.” <em>Molecular &amp; Cellular Proteomics : MCP</em> 4 (12): 1920–32. <a href="https://doi.org/10.1074/mcp.M500279-MCP200">https://doi.org/10.1074/mcp.M500279-MCP200</a>.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>www.proteinatlas.org<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Laurent Gatto, Manon Martin.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
